src/snmp.c

```diff
@@ -449,6 +451,48 @@ static int csnmp_config_add_host_version(host_definition_t *hd,
   return 0;
 } /* int csnmp_config_add_host_address */
 
+static int csnmp_config_add_host_timeout(host_definition_t *hd,
+                                         oconfig_item_t *ci) {
+  int timeout;
+
+  if ((ci->values_num != 1) || (ci->values[0].type != OCONFIG_TYPE_NUMBER)) {
+    WARNING("snmp plugin: `Timeout' must be a number");
+    return -1;
+  }
+
+  timeout = (int)ci->values[0].value.number;
+  if (timeout < 0) {
+    WARNING("snmp plugin: `Timeout' must not be negative");
+    return -1;
+  }
+
+  /* net-snmp library timeout is in microseconds */
+  hd->timeout = timeout * 1000000;
```

Comment:
Actually, with this: definitely change `hd->timeout` to be a `cdtime_t` and use `cf_util_get_cdtime()` to parse the config. You can then use `CDTIME_T_TO_US(hd->timeout)` to convert it to the µs expected by net-snmp.